flink-core/src/main/java/org/apache/flink/api/common/TaskInfo.java

```diff
@@ -71,7 +71,7 @@ public TaskInfo(
 		this.indexOfSubtask = indexOfSubtask;
 		this.numberOfParallelSubtasks = numberOfParallelSubtasks;
 		this.attemptNumber = attemptNumber;
-		this.taskNameWithSubtasks = taskName + " (" + (indexOfSubtask + 1) + '/' + numberOfParallelSubtasks + ')';
+		this.taskNameWithSubtasks = taskName + " (" + (indexOfSubtask + 1) + '/' + numberOfParallelSubtasks + ')' + " - execution #" + attemptNumber;
```

Comment:
In order to not blow up the length of the messages too much, maybe just append ` + "#" + attemptNumber`? After all:
- other components are not named
- reminding million times in a single log that this is `execution` seems an overkill ;)

Reply:
done.